### PR TITLE
cursor: send clear_focus on hide and enter event on unhide

### LINKED ISSUE #3431: Send wl_pointer_leave when hide_cursor activates
So that e.g. browsers don't show tooltips.

### PR DESCRIPTION
**Pull request by Emantor**
Clear the focus when we hide the cursor and show it again during the unhide
action. The unhide function will send an enter event to the currently focused
window.

Tested by looking at the WAYLAND_DEBUG=1 output of termite.

Fixes #3431

### CODE REVIEW

**Comment by Emantor** [comment]
Looking at the code again we sent the enter event after we move the cursor. Are there edge cases where this can be a problem?

**Comment by Emantor** [comment]
I still have input issue in firefox in xwayland, not sure if thats a bug introduced by this PR or an xwayland bug...

File: sway/input/cursor.c [hunk 1]
```diff
@@ -313,6 +314,13 @@ void cursor_unhide(struct sway_cursor *cursor) {
 		cursor->image = NULL;
 		cursor_set_image(cursor, image, cursor->image_client);
 	}
+	struct sway_node *focus = seat_get_focus(cursor->seat);
```
Comment (RedSoxFan) [inline sway/input/cursor.c]: I think this should be replaced with `cursor_rebase(cursor)` so that it handles layer surfaces as well

**Comment by emersion** [comment]
>I still have input issue in firefox in xwayland, not sure if thats a bug introduced by this PR or an xwayland bug...

This bug also happens when switching workspaces, so I don't think it's related.

File: sway/input/cursor.c (hunk 1 above)
Comment (Emantor) [reply sway/input/cursor.c]: Nice catch, this fixed the input bug I was experiencing.

**Comment by RedSoxFan** [comment]
Thanks!

**Pull request merged**